src/Examples/DAE/petsc_pid_example.ipynb

```diff
@@ -666,20 +695,347 @@
     }
    ],
    "source": [
-    "a = plt.plot(tj.vecs[\"_time\"], tj.vecs[str(m.fs.valve_1.control_volume.properties_in[24].pressure)])\n",
-    "a = plt.ylabel(\"inlet pressure (Pa)\")\n",
+    "a = plt.plot(tj.time, tj.get_vec(m.fs.tank.control_volume.properties_out[24].pressure))\n",
+    "a = plt.ylabel(\"tank pressure (Pa)\")\n",
     "a = plt.xlabel(\"time (s)\")"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "aa792db6-9edf-4c42-b77a-5b50672c4362",
+   "metadata": {},
+   "source": [
+    "## Model a ramp in inlet pressure (again)\n",
+    "\n",
+    "Here we repeat the ramp from the previous simulation in a differnt way.  In this case we do the integration in three parts. 1) Constant pressure at 500 kPa to 10 s 2) ramp from 500 to 600 kPa from 10 to 12 s. 3) Constant pressure at 600 kPa from 12 to 24 s."
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": 12,
-   "id": "3f6429ec-1c35-4626-98c4-2577126d5778",
+   "id": "e2fc8180-9ad7-4cf5-9aad-78b882730347",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2022-04-21 14:41:35 [INFO] idaes.init.fs.valve_1: Initialization Complete: optimal - Optimal Solution Found\n",
+      "2022-04-21 14:41:35 [INFO] idaes.init.fs.tank.control_volume: Initialization Complete\n",
+      "2022-04-21 14:41:35 [INFO] idaes.init.fs.tank: Initialization Complete: optimal - Optimal Solution Found\n",
+      "2022-04-21 14:41:35 [INFO] idaes.init.fs.valve_2: Initialization Complete: optimal - Optimal Solution Found\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Create a new copy of the model that runs to 24 seconds, and add a constraint.\n",
+    "\n",
+    "m = pid.create_model(\n",
+    "    time_set=[0, 10, 12, 24],\n",
+    "    nfe=3,\n",
+    "    calc_integ=True,\n",
+    ")\n",
+    "# time_var will be an explicit time variable we can use in constraints.\n",
+    "m.fs.time_var = pyo.Var(m.fs.time)\n",
+    "for t in m.fs.time:\n",
+    "    m.fs.time_var[t].set_value(t)\n",
+    "\n",
+    "# We'll add a constraint to calculate the inlet pressure based on time,\n",
+    "# so we need to unfix pressure.\n",
+    "m.fs.valve_1.control_volume.properties_in[0].pressure.fix(500000)\n",
+    "m.fs.valve_1.control_volume.properties_in[10].pressure.fix(500000)\n",
+    "m.fs.valve_1.control_volume.properties_in[12].pressure.set_value(600000)\n",
+    "m.fs.valve_1.control_volume.properties_in[12].pressure.unfix()\n",
+    "m.fs.valve_1.control_volume.properties_in[24].pressure.fix(600000)\n",
```

Comment:
This comment seems to conflict with the code. Why is the pressure being fixed at 3 of the points and unfixed at 1?